app/routes/progress-saved.js

```diff
@@ -0,0 +1,16 @@
+const cache = require('../cache')
+const { save } = require('../agreement')
+
+module.exports = {
+  method: 'POST',
+  path: '/progress-saved',
+  options: {
+    handler: async (request, h) => {
+      const { agreement } = await cache.get(request)
+      const agreementNumber = agreement.agreementNumber
+      const referer = request.headers.referer ? new URL(request.headers.referer).pathname : '/'
+      await save(request)
+      return h.view('progress-saved', { referer, agreementNumber })
+    }
+  }
+}
```

Comment:
I would maybe add this to the existing `./routes/save.js` module as that is already doing the same thing, albeit with a get.

Also following a post, should redirect to a get to return the view.  Otherwise a user refreshing the page will keep resubmitting the data over and over.

I think keeping the name `/save` is a better description too or maybe `save-progress` as the past tense of this implies the save has already happened.